src/lualib/ArrayReduceRight.ts

```diff
@@ -3,7 +3,7 @@ function __TS__ArrayReduceRight<T>(
     this: void,
     arr: T[],
     callbackFn: (accumulator: T, currentValue: T, index: number, array: T[]) => T,
-    ...initial: Vararg<T>
+    ...initial: Vararg<T[]>
```

Comment:
This does not seem correct, VarArg<T> extends Array<T>, so seems like initial is T[][] here?

Reply:
I have changed `Vararg` declaration (https://github.com/TypeScriptToLua/TypeScriptToLua/pull/839/files#diff-a325139454ae742b51dd67eea598d304R5), since it was triggering [no-empty-interface](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-interface.md). This definition of `Vararg` is probably going to be added to builtin types anyway